src/chainrules.jl

```diff
@@ -193,3 +193,66 @@ ChainRulesCore.@scalar_rule(
 ChainRulesCore.@scalar_rule(expinti(x), exp(x) / x)
 ChainRulesCore.@scalar_rule(sinint(x), sinc(invπ * x))
 ChainRulesCore.@scalar_rule(cosint(x), cos(x) / x)
+
+# non-holomorphic functions
+function ChainRulesCore.frule((_, _, _), ::typeof(besselix), ν::Number, x::Number)
+    Ω = besselix(ν, x)
+    ΔΩ = ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO)
+    return Ω, ΔΩ
+end
+function ChainRulesCore.rrule(::typeof(besselix), ν::Number, x::Number)
+    Ω = besselix(ν, x)
+    project_x = ChainRulesCore.ProjectTo(x)
+    function besselix_pullback(ΔΩ)
+        ν̄ = ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO)
+        a = (besselix(ν - 1, x) + besselix(ν + 1, x)) / 2
```

Comment:
I just reused the derivatives that are used for `besseli` etc. They were defined in this way in ChainRules originally. When I copied them to SpecialFunctions I wondered about the motivation for choosing https://functions.wolfram.com/Bessel-TypeFunctions/BesselI/20/01/02/0003/ over https://functions.wolfram.com/Bessel-TypeFunctions/BesselI/20/01/02/0001/ or https://functions.wolfram.com/Bessel-TypeFunctions/BesselI/20/01/02/0002/. I assumed the currently used definition is simpler since it does not require to handle `x = 0` in a special way.

Reply:
I think it might be best to use the same relations for `besselix` etc. as for `besseli` etc. and, if desired, change them to a different form in a separate PR.

Reply:
That's right, and they're the same in DiffRules as well. If you like, you can keep them similar to `besseli`, etc for now, and a future PR could update all of the rules.

Reply:
I agree!